src/dotnet-compile-fsc/Program.cs

```diff
@@ -295,5 +278,100 @@ private static IEnumerable<string> GetSourceFiles(IReadOnlyList<string> sourceFi
 
             yield return sourceFiles.Last();
         }
+
+        private static Command RunFsc(List<string> fscArgs, string temp)
+        {
+            var fscEnvExe = Environment.GetEnvironmentVariable("DOTNET_FSC_PATH");
+            var exec = Environment.GetEnvironmentVariable("DOTNET_FSC_EXEC")?.ToUpper() ?? "COREHOST";
+            
+            var muxer = new Muxer();
+
+            if (fscEnvExe != null)
+            {
+                switch (exec)
+                {
+                    case "RUN":
+                        return Command.Create(fscEnvExe, fscArgs.ToArray());
+
+                    case "COREHOST":
+                    default:
+                        var host = muxer.MuxerPath;
+                        return Command.Create(host, new[] { fscEnvExe }.Concat(fscArgs).ToArray());
+                }
+            }
+            else
+            {
+                var fscCommandSpec =  ResolveFsc(fscArgs, temp)?.Spec;
+                return Command.Create(fscCommandSpec);
```

Comment:
After FSC makes changes to include runtimeconfig.json and put win32manifest in some other location, this can use the commandFactory. 

This is because dotnet-compile-fsc needs to get the path to fsc.exe which isn't provided through the factory interface.